File: src/data/migrations/0002_addsessionsettings.cpp
#include "0002_addsessionsettings.hpp"

#include <boost/log/trivial.hpp>

using porla::Data::Migrations::AddSessionSettings;

int AddSessionSettings::Migrate(sqlite3* db)
{
    BOOST_LOG_TRIVIAL(info) << "Creating 'sessionsettings' table";

    return sqlite3_exec(
        db,
        "CREATE TABLE sessionsettings ("
        "id INTEGER PRIMARY KEY,"
        "key TEXT NOT NULL,"
        "value TEXT NOT NULL,"
        "UNIQUE (key)"
        ");",
        nullptr,
        nullptr,
        nullptr);
}

File: lua/db/migrations/0002_add_session_settings.lua
local M = {}

function M.migrate(db)
    print("Creating 'sessionsettings' table")

    db:exec([[
        CREATE TABLE sessionsettings (
            id INTEGER PRIMARY KEY,
            key TEXT NOT NULL,
            value TEXT NOT NULL,
            UNIQUE (key)
        );
    ]])
end

return M
